### PR TITLE
Pdf color none

### PR DESCRIPTION
**Pull request by tacaswell**
Replaces #8674

### CODE REVIEW

File: lib/matplotlib/tests/test_backend_pdf.py [hunk 1]
```diff
@@ -191,3 +193,17 @@ def psfont(*args, **kwargs):
     ax.text(0.5, 0.5, 'hello')
     with tempfile.TemporaryFile() as tmpfile, pytest.raises(ValueError):
         fig.savefig(tmpfile, format='pdf')
+
+
+def test_pdf_savefig_when_color_is_none(tempdir_factory):
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_backend_pdf.py]: The simpler `tmpdir` would be better here; don't see why it needs a factory.

File: lib/matplotlib/tests/test_backend_pdf.py [hunk 2]
```diff
@@ -191,3 +193,17 @@ def psfont(*args, **kwargs):
...
+    rcParams['_internal.classic_mode'] = False
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_backend_pdf.py]: I don't know what this entails? It's neither classic nor default style now.

File: lib/matplotlib/tests/test_backend_pdf.py (hunk 1 above)
Comment (anntzer) [inline lib/matplotlib/tests/test_backend_pdf.py]: I don't even see where tempdir_factory is defined?

Comment (QuLogic) [inline lib/matplotlib/tests/test_backend_pdf.py]: Probably a typo, which is why it's not passing; should have been [`tmpdir_factory`](https://docs.pytest.org/en/latest/tmpdir.html#the-tmpdir-factory-fixture).

File: lib/matplotlib/tests/test_backend_pdf.py [hunk 3]
```diff
@@ -191,3 +193,17 @@ def psfont(*args, **kwargs):
     ax.text(0.5, 0.5, 'hello')
     with tempfile.TemporaryFile() as tmpfile, pytest.raises(ValueError):
         fig.savefig(tmpfile, format='pdf')
+
+
+@pytest.mark.style('default')
+def test_pdf_savefig_when_color_is_none(tmpdir_factory):
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_backend_pdf.py]: The `tmpdir` fixture is simpler...

Comment (tacaswell) [inline lib/matplotlib/tests/test_backend_pdf.py]: but then I have to sort out how to use it....

Comment (QuLogic) [inline lib/matplotlib/tests/test_backend_pdf.py]: Just drop the `.mktemp()` since it's doing that for you.

**Comment by tacaswell** [comment]
Squashed commits, simplified tests, and changed to use public API instead of private attributes.

File: lib/matplotlib/testing/compare.py [hunk 4]
```diff
@@ -355,8 +355,8 @@ def crop_to_same(actual_path, actual_image, expected_path, expected_image):
     # clip the images to the same size -- this is useful only when
     # comparing eps to pdf
     if actual_path[-7:-4] == 'eps' and expected_path[-7:-4] == 'pdf':
-        aw, ah = actual_image.shape
-        ew, eh = expected_image.shape
+        aw, ah, ad = actual_image.shape
```
Comment (anntzer) [inline lib/matplotlib/testing/compare.py]: What happened here?  How would this have worked before?

Comment (tacaswell) [inline lib/matplotlib/testing/compare.py]: According to coverage it was not run before https://codecov.io/gh/matplotlib/matplotlib/compare/f5f3d793a7a14cfd186f76a987826e58c3337437...51f070536cdb8b89552f03091b3068a6fc45e5a0/src/lib/matplotlib/testing/compare.py#L357

**Comment by tacaswell** [comment]
I am :+1: on merging this, but can not review a PR I opened (which sorta makes sense).

Anyone else who approves this should merge it.

**Comment by tacaswell** [comment]
@Tuan333 Thanks for you work on this!

**Pull request merged**